doc: Update the CONTRIBUTING.md to follow Conventional Commits (#16823)
apache/nuttx: CONTRIBUTING.md

 8. Change must be well documented (build / runtime test logs, pr, git
    commit, documentation, release notes, etc) with clear notes on how to
    fix the introduced problems.

>  9. Breaking Change must be clearly identified with inclusion of the an

Reviewer: 9. Breaking Changes must be clearly marked both in Git Commits and Pull Requests: 1. Put exclamation mark `!` as first character of the title; 2. Put `BREAKING CHANGE:` in the body with a brief change description and the quick-fix instructions. This helps users identify areas that need an update on their side, grep git logs, and auto-generate Release Notes.

@hartmannathan could you please verify the grammar? :-P

>  9. Breaking Change must be clearly identified with inclusion of the an
>     exclamation mark `!` as first character of the commit title and a
>     `BREAKING CHANGE:` in the git commit log message, followed by a short
>     comment about way user will use it currently. This change description
>     will be used to create the Release Notes.

Reviewer 2: Suggestion:

```suggestion
 9. Breaking change must be clearly identified with inclusion of an
    exclamation mark `!` as the first character of the commit title and a
    `BREAKING CHANGE:` in the git commit log message, followed by a short
    comment about how the user should adapt their code. This change
    description will be used to create the Release Notes.
```

Thoughts?

Reviewer: Thanks @hartmannathan :-)
* what is the difference between "identified" and "marked"? is it author who "marks" and recipient who "identifies"? are IC marked on the casing or identified on the casing? :-P
* Pull Requests are omitted here, we should keep Git Commits and Pull Requests tied with the same rule. Release Notes are built from PR titles. This allows separation of breaking changes in the Release Notes. This saves time for the user after update. Git Logs are first contact to see why things broke / how things changed, contain more details than release notes, thus my suggestion git log can be simply grepped / searched for specific marks to find fixes.
* I like the idea that we should somehow require authors to provide a brief overview on what changed with a quick-fix, using this "BREAKING CHANGE:" mark, if something needs an update and the user is not an expert, also to have consistent fixes implemented. Details of the change will be bundled in the description too, but its more about what changed and how to fix if broken. "Adopt their code" is nice but too wide does not imply a specific fix. I know there is no quick fix always possible, then a hint is enough. Should we make it more precise?
* How to update last sentence to show that not only release notes are built that way but also git log can be searched for hints and fixes after update?

Breaking changes happen and we should do best we can to prepare users for smooth updates :-)

Not all changes will affect all users and not all users are experts so we should provide braindead simple way to fix things :-)
